text/0059-token-metadata-document.md

```diff
@@ -0,0 +1,504 @@
+
+We take the point of view that these differences are not "token metadata", which are determined by the ID of the token that we are using, but rather localization properties determined by the locale of the user. So, for example the same app might be capable of displaying `kr 127,54` and `127.54 kr`, depending on the locale of the user, and token metadata does not influence this.
+
+This is also consistent with [how shopify handles currency localization](https://polaris.shopify.com/foundations/formatting-localized-currency).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The token-metadata json fields have the following semantics:
+
+* `schema_version`: An integer describing the version number of the schema of this document. This is 1 until a future MCIP proposes schema modifications.
+* `signing_timestamp`: A string representation of the unix timestamp at the time of creating the document's signature.
+* `token_id`: A decimal string representation of the `u64` corresponding to the token id.
+* `currency_name`: A UTF-8 string which is the full, official name of the currency. This is intended to be an english-language noun or noun phrase.
+* `ticker_symbol`: A ticker symbol for the currency. This is appropriate to be displayed by an exchange, as a short form of the name of the currency. Not more than 12 printable ASCII characters, including `[A-Za-z0-9].-`.
+* `symbol`: Most fiat currencies have a printable character such as $, £, ¥. Some cryptocurrencies do also, Bitcoin has ₿, and Ethereum has Ξ. This may be optionally specified as a UTF-8 string in the "symbol" field. How exactly it is displayed may be locale specific and we do not attempt to formally specify that at this time.
+* `decimals`: An integer specifying how the `u64` integer in a TxOut in the blockchain is scaled to compute a user-displayed amount of the currency.
+* `logo_svg`: An optional logo image. This is a base64-encoded SVG document. This is expected to have been sanitized using something like [svg-hush](https://github.com/cloudflare/svg-hush), to remove scripting, hyperlinks to other documents, and references to cross-origin resources. The full extent of such sanitization will not be specified here.
+* `info_url`: A link to a website containing more information about the token that may be interesting to token holders. This should basic information about the purpose of the token, its supply, any utility that it has, or links to associated whitepapers.
+
+Clients MUST download and validate the `token_metadata.sig` before attempting to process the `token_metadata.json`, and must reject the json with an error if the signature fails.
```

Comment:
This may need more detail on _how_.
- Is it only the non whitespace characters being signed, i.e. minified JSON
- Is it the JSON file as presented.  If it's as presented may want to specify '\n' line endings for those that work on windows where '\r\n' is the usual line ending.

Reply:
I think the signature is over whatever bytes are inside the json file. Whether it has `\n` or `\r\n` in it for line endings should not matter.

Reply:
:+1: I think it's clearer.  

There is a chance it fails to validate if a tool replaces line endings or trailing newline during the download process, but I'm not sure how to prevent that without a giant note here that might get in the way.  

It's probably best to wait and see if it's enough of an issue to justify a note or similar in docs.  The worse that happens is someone isn't able to verify the signature.

Reply:
> There is a chance it fails to validate if a tool replaces line endings or trailing newline during the download process, but I'm not sure how to prevent that without a giant note here that might get in the way.

I don't think this is likely to happen, the document is going to be the body in an HTTP GET response. Curl does not modify this body, and no other standard http library is going to modify the bytes of the body in a way that is unavoidable. Http libraries provide access to the bytes of the body of the HTTP response.